### PR TITLE
Add test case order validation

### LINKED ISSUE #153: Multiple testcase bug
A old testcase is always being displayed although there are newer ones. The correct behaviour would be to display the newest testcase in both the game page and the main page.

An example would be this: https://citra-emu.org/game/pokemon-super-mystery-dungeon/

### PR DESCRIPTION
**Pull request by BynariStar**
Edit: #153 is closed by citra-emu/citra-web#64 instead.

### CODE REVIEW

File: validation/app.js [hunk 1]
```diff
@@ -252,6 +252,13 @@ function validateTOML(path) {
             validateNotEmpty(testcase, "gpu");
             validateNotEmpty(testcase, "os");
         });
+        
+        //Validate dates are properly ordered
+        section.reduce(function(a, v){
```
Comment (j-selby) [inline validation/app.js]: These could be much more creatively named. Further, space between `v)` and `{` please.

File: validation/app.js [hunk 2]
```diff
@@ -252,6 +252,13 @@ function validateTOML(path) {
...
+            if (typeof a === "undefined" || a.date <= v.date)
```
Comment (j-selby) [inline validation/app.js]: for consistencies sake, make this a block:
```javascript
...) {
    return v;
}
```

File: validation/app.js [hunk 3]
```diff
@@ -252,6 +252,13 @@ function validateTOML(path) {
             validateNotEmpty(testcase, "gpu");
             validateNotEmpty(testcase, "os");
         });
+        
+        //Validate dates are properly ordered
```
Comment (j-selby) [inline validation/app.js]: Space between `//` and `Validate`.

**Comment by j-selby** [comment]
Hold up - this isn't detecting current bugs in the ordering of these entries (see: pokemon super mystery dungeon). Are you sure this is working?

**Comment by BynariStar** [comment]
If you look at the commits up until I changed the order to ascending they didn't pass Travis for this reason. However it was the decided that the order format of the test cases are fine and that instead Hugo should just display them backwards,  but that's for another PR. However this change is still useful in the event that someone submits a test case in an order that is different than the current one.

**Comment by j-selby** [comment]
Ah, derp. Should read next time :P
All good!

**Pull request merged**